src/coordinator/parser/promql/types.go

```diff
@@ -103,11 +99,41 @@ func NewFunctionExpr(name string, argValues []interface{}) (parser.Params, error
 		return linear.NewMathOp(linear.Log10Type)
 	case linear.Log2Type:
 		return linear.NewMathOp(linear.Log2Type)
-	case linear.RoundType:
-		return linear.NewRoundOp(argValues)
 	case linear.SqrtType:
 		return linear.NewMathOp(linear.SqrtType)
 
+	case linear.AbsentType:
+		return linear.NewAbsentOp(), nil
+
+	case linear.ClampMinType:
+		return linear.NewClampOp(argValues, linear.ClampMinType)
+	case linear.ClampMaxType:
+		return linear.NewClampOp(argValues, linear.ClampMaxType)
+
+	case linear.RoundType:
+		return linear.NewRoundOp(argValues)
+
+	case datetime.DayOfMonthType:
```

Comment:
all this can be a single case:

case datetime.DayOfMonthType, datetime.DayOfWeekType, ....: 
   return datetime.NewDateOp(name)

Reply:
Also do it for other things (I created a bad pattern here)